### PR TITLE
ATO-426 0 Enabling CTest for subset of the STAT classes

### PR DESCRIPTION
**Pull request by miranov25**
### Modifications
*  testAliTreePlayer
```
ctest -R testAliTreePlayer --verbose
95: I-AliTreePlayerTest.testConvertTree: Test OK
95: Processing /u/miranov/alicesw/sw/SOURCES/AliRoot/alicesw/0/STAT/test/AliTreePlayerTest.C+...
95: I-AliTreePlayerTest.testSelectMetadata AND invariant: Test OK: N(Logbook&&(Stat))=N(Logbook&&(Stat&&Base))+N(Logbook&&(Stat&&(!Base)))       7=2+5
95: I-AliTreePlayerTest.testSelectMetadata Negatioation test: Test OK:   N(!(A||B))=N((!A)&&(!B)) 73==73
95: I-AliTreePlayerTest.testselectTreeInfo: Test OK:   N(A))==N(A&&B)&&N(A&&!B)) 46=6+40
95: I-AliTreePlayerTest.testConvertTree: Test OK
95: [2017-07-12 15:26:50 CEST] [lxbk0198] [SUCCESS] statTest.testAliTreePlayer: All OK 
1/1 Test #95: testAliTreePlayer ................   Passed    1.82 sec

The following tests passed:
        testAliTreePlayer

100% tests passed, 0 tests failed out of 1
```
*  testAliTMinutiToolkitTestLinear
```
ctest -R  testAliTMinuitToolkitTest --verbose
...
96: I-AliTMinuitToolkitTestLinear: FitLogLike:Gaus(80%)Cauchy(20%)- MISAC(50)
96: I-TCanvas::Print: png file AliTMinuiToolkit.TestLinearFitStatExample.png has been created
96: I-TCanvas::Print: pdf file AliTMinuiToolkit.TestLinearFitStatExample.pdf has been created
96: [2017-07-12 15:15:49 CEST] [lxbk0198] [SUCCESS] statTest.testAliTMinutiToolkitTestLinear: All OK 
1/1 Test #96: testAliTMinuitToolkitTest ........   Passed    7.63 sec
The following tests passed:
        testAliTMinuitToolkitTest
100% tests passed, 0 tests failed out of 1
```
### Tests
* Ubuntu system (laptop) 
  * root 5 - OK
  * root 6 - OK
* Debian (GSI batch farm)
  * root 5

### HTTP access to QA server needed 
* test on server to be done
  * According discussion with Peter test server should have an access

### CODE REVIEW

**Comment by miranov25** [comment]
hallo @pzhristov, @dberzano, @ktf 

I'm starting to include test of selected classes in the STAT directory into the CTEST tools.
I started with 2 tests (see test results on different platforms above).  
In have another  4 tests in queue. Most important will be tests of elastic search interface
and AliExternalInfo class

Marian

**Comment by dberzano** [comment]
Hi @miranov25, can you make the pull request "editable" by me? On the left side of the GitHub interface you should see something like "make editable by maintainers". I'd like to fix, directly, some very minor things (just syntax).

Thanks for the test!

**Comment by miranov25** [comment]
Hallo @dberzano 
Allow edits from maintainers. is checked.
Is there some problem?
Maybe you can show me on Vydio what to do?

**Comment by dberzano** [comment]
I'll modify directly and you'll see (it's much much faster).

**Comment by miranov25** [comment]
@dberzano @ktf @pzhristov @mkrzewic 
Do you know by chance somebody who used grok/logstash
I would like to integrate grok parser, to parse  log files to json structured message lines.

**Comment by dberzano** [comment]
Hi @miranov25, I did some cleanup in the shell script and the CMake files (the rest is just comments and whitespaces). Tests have been uniformed with the rest of our tests, and some environment hell has been cleaned up. Any problem with that?

**Comment by miranov25** [comment]
Hallo Dario.

Thanks for your time. I checked your modifications. Many of them were fine and they indeed help readability and uniformity. But there is a couple of others I'm concerned about:

#### My consideration:
* Log files   - we need them
  * additional step parsing of log files creating json summary documents (e.g for elastic) is on the way
  * I prefer to keep/archive log files for some time
  * maybe not for each build but on regular basis
* working in tmp directories and cleaning
  *  I consider working directory should be cleaned/resp archived/indexed  centrally
    * parameterization of beahaviour should be central (switch/variable)
  *  Generic case: if something fails I would like to check diff the content
    * I do not want to modify the test scripts in order to be able to make such check
* in many cases you removed comments which were indeed irrelevant
  * OK
  * I consider comments describing how to run the code relevant
  * I have tendency to forget exact sequence of steps. Keeping this informtation in the commented field help significantly
* change of the comment style in C,C++
  * thanks to the CINT/CLANG i can copy/paste and test segments of the code in root session
  * using /// coments instead of the commenting blocks prevents copy/paste
    * I will revert this changes
* using path in commented example code
  * In all my macros I'm using consistently  $AliRoot_SRC and $AliPhysics_SRC variables in comment code snippets. These variables I define when loading AliRoot/AliPhysics
  * Writing the code/returning back to my code I  simply do copy/paste of commented code snippet
  * I do not use directly $ALICE_ROOT $ALICE_PHSYSICS as I do not want to make install after each change 
  * We can consider renaming of the <Alixxx>_SRC variables to something else, but modification should still make copy/paste of code snippets possible
  * This changes I prefer to revert

How do we proceed? 
* I can merge part of your modification into my code now or we can clarify basic assumptions.
* Was your code after modification working on the test server

**Comment by dberzano** [comment]
@miranov25 can you be more concise in your comment? I am quite busy and don't have time to read all of it, sorry (nor to discuss it in person). What I did is consistent with the rest of the tests.

**Comment by miranov25** [comment]
* I tried to explain which changes I can accept and which one I can not. 
  * I explained in previous comment what is the motivation
* Concerning consistency with old test: I would like to extend CTest and use it also for regression test - exporting some variables to DB and keeping log files 
* Before partial merge of your modification I wanted to know if it will be accepted to avoid unnecessary iterations.

**Comment by pzhristov** [comment]
+test

**Comment by miranov25** [comment]
TLDR: I reverted some of your changes, because they break my current workflow. Get back to me when you have more time to discuss

**Comment by dberzano** [comment]
Please comply as opposed to reverting otherwise we will never approve this. We have a standard for running tests, according to your requests minor modifications (not reverts) are needed. We're trying to get to a compromise here, but since we have a standard for running tests we will not allow you to not follow it, sorry.

**Comment by miranov25** [comment]
I meant I revert back changes which are against workflow which I use. 
Other changes which you implemented and which were indeed relevant I will keep

**Comment by miranov25** [comment]
hallo @pzhristov, @dberzano and all.

I reviewed proposed changes by @dberzano. I included small modifications as I described in previous comments.

* Using code blocks comments  instead of the line comments.
*  I added doxygen documentation for the STAT/test directory. 
  * doxygen nicely process code blocks comments in the AliTreePlayerTest.C
* I keep log files for later processing (Elastic indexing, regression test)

Output of ctest using updated code:
```
ctest -R func_STAT_*  --extra-verbose
UpdateCTestConfiguration  from :/data/alicesw/sw/BUILD/AliRoot-latest/AliRoot/DartConfiguration.tcl
UpdateCTestConfiguration  from :/data/alicesw/sw/BUILD/AliRoot-latest/AliRoot/DartConfiguration.tcl
Test project /data/alicesw/sw/BUILD/AliRoot-latest/AliRoot
Constructing a list of tests
Done constructing a list of tests
Checking test dependency graph...
Checking test dependency graph end
test 96
    Start 96: func_STAT_AliTreePlayer

96: Test command: /bin/bash "-c" "ALIROOT_SOURCE=/data/alicesw/sw/SOURCES/AliRoot/alicesw/0 /data/alicesw/sw/SOURCES/AliRoot/alicesw/0/STAT/test/statTest.sh testAliTreePlayer"
96: Test timeout computed to be: 9.99988e+06
96: [2017-07-14 06:59:15 CEST] [ivanov-laptop] [SUCCESS] statTest.testAliTreePlayer: All OK 
1/2 Test #96: func_STAT_AliTreePlayer ...........   Passed    2.46 sec
test 97
    Start 97: func_STAT_AliTMinuitToolkitTest

97: Test command: /bin/bash "-c" "ALIROOT_SOURCE=/data/alicesw/sw/SOURCES/AliRoot/alicesw/0 /data/alicesw/sw/SOURCES/AliRoot/alicesw/0/STAT/test/statTest.sh testAliTMinutiToolkitTestLinear"
97: Test timeout computed to be: 9.99988e+06
97: [2017-07-14 06:59:19 CEST] [ivanov-laptop] [SUCCESS] statTest.testAliTMinutiToolkitTestLinear: All OK 
2/2 Test #97: func_STAT_AliTMinuitToolkitTest ...   Passed    4.53 sec

The following tests passed:
        func_STAT_AliTreePlayer
        func_STAT_AliTMinuitToolkitTest

100% tests passed, 0 tests failed out of 2

Total Test time (real) =   7.01 sec
```

**Comment by miranov25** [comment]
Hallo @pzhristov 
could you please approve test?  Local tests are running. Changes above were minimal.

**Comment by pzhristov** [comment]
+test

**Comment by miranov25** [comment]
@dberzano, @pzhristov  
Could you comment on pull request?

**Comment by miranov25** [comment]
Yes/No/Later?

**Comment by dberzano** [comment]
It's 6pm and it's Friday evening. Next week I'll review it. Have a nice weekend!

**Comment by dberzano** [comment]
Hi @miranov25, we can find a time to meet, I wanted to ask you too. (But let's discuss the details offline please.)

Let me publicly comment on your sentence, hoping it's the last time I have to do this:

> In case of not getting pull request today, I will continue with implementation of other tests on my github

This attitude is utterly disrespectful of other people's work. I am responsible for this part of testing, and welcome any contribution (we like tests!), but this part is not mission critical, therefore I see no urgency (_i.e._ TODAY) of having it in. Remember, there's a collaboration out there, not just you and your needs.

* You come out of the blue with a PR. And you require it to be merged ASAP. I say, we have tons of work to do, and I even like where your PR is going, but it's my task, job, and responsibility, to make sure we (me and you) both like it. This takes time (only a little, but we have plenty of littles-by-littles here) and will to compromise.  And since you are not my boss (@pzhristov is), you do not get to decide _when_ I review your PR. Which will be done by Thursday in any case, but not today.

* I take no threats, especially in the form "I will continue my way anyways if you don't merge". Keep going with this attitude and you will need your own repository where you can do whatever you want. Feel free to escalate the problem if you want, we have an Offline week ahead.

That said, please do not let us unbury our hatchets. We like peace, peace is good :-)

**Comment by dberzano** [comment]
+1
